plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -1959,6 +1959,78 @@ public void testHistoryTable()
         }
     }
 
+    @Test
+    public void testHistoryTableWithDeletedTransactionLog()
+            throws Exception
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_history_table_with_deleted_transaction_log",
+                "(int_col INTEGER)")) {
+
+            for (int i = 1; i <= 8; i++) {
+                assertUpdate("INSERT INTO " + table.getName() + " VALUES (" + i + ")", 1);
+            }
+            // Add more transaction to generate checkpoint file
+            assertUpdate("DELETE FROM " + table.getName() + " WHERE int_col = 1", 1);
+            assertUpdate("UPDATE " + table.getName() + " SET int_col = int_col * 2 WHERE int_col = 6", 1);
+
+            assertQuery(
+                    "SELECT version, operation FROM \"" + table.getName() + "$history\"",
+                    "VALUES (0, 'CREATE TABLE'), (1, 'WRITE'), (2, 'WRITE'), (3, 'WRITE'), (4, 'WRITE'), (5, 'WRITE'), (6, 'WRITE'), (7, 'WRITE'), (8, 'WRITE'), (9, 'MERGE'), (10, 'MERGE')");
+            assertQuery(
+                    "SELECT version, operation FROM \"" + table.getName() + "$history\" WHERE version = 8",
+                    "VALUES (8, 'WRITE')");
+            assertQuery(
+                    "SELECT version, operation FROM \"" + table.getName() + "$history\" WHERE version > 8",
+                    "VALUES (9, 'MERGE'), (10, 'MERGE')");
+            assertQuery(
+                    "SELECT version, operation FROM \"" + table.getName() + "$history\" WHERE version >= 8 OR version = 1",
+                    "VALUES (1, 'WRITE'), (8, 'WRITE'), (9, 'MERGE'), (10, 'MERGE')");
+            assertQuery(
+                    "SELECT version, operation FROM \"" + table.getName() + "$history\" WHERE version >= 1 AND version < 3",
+                    "VALUES (1, 'WRITE'), (2, 'WRITE')");
+            assertQuery(
+                    "SELECT version, operation FROM \"" + table.getName() + "$history\" LIMIT 1",
+                    "VALUES (0, 'CREATE TABLE')");
+
+            assertThat(query("SELECT version, operation FROM \"" + table.getName() + "$history\" WHERE version > 1 AND version < 2"))
+                    .returnsEmptyResult();
+            assertThat(query("SELECT count(*) FROM \"" + table.getName() + "$history\""))
+                    .matches("VALUES bigint '11'");
+
+            String tableLocation = getTableLocation(table.getName());
+
+            // Remove first two transaction logs
+            String key = tableLocation.substring(bucketUrl().length());
+            MinioClient minio = hiveMinioDataLake.getMinioClient();
+            minio.removeObject(bucketName, "%s/_delta_log/%020d.json".formatted(key, 0));
```

Comment:
Is `delta.logRetentionDuration` , `register_table` and delta lake OSS maybe of help here to simulate a more productive scenario? See https://github.com/trinodb/trino/issues/16207

Reply:
As discussed offline, to simulate the log retention period, we will remove the transaction log.